test/utils/ansible-test/lib/classification.py

```diff
@@ -0,0 +1,321 @@
+
+    commands = dict((c, sorted(commands[c])) for c in commands.keys() if commands[c])
+
+    return commands
+
+
+class PathMapper(object):
+    """Map file paths to test commands and targets."""
+    def __init__(self):
+        self.integration_targets = list(walk_integration_targets())
+        self.module_targets = list(walk_module_targets())
+        self.compile_targets = list(walk_compile_targets())
+        self.units_targets = list(walk_units_targets())
+
+        self.compile_paths = set(t.path for t in self.compile_targets)
+        self.units_modules = set(t.module for t in self.units_targets if t.module)
+        self.units_paths = set(t.path for t in self.units_targets)
+
+        self.module_names_by_path = dict((t.path, t.module) for t in self.module_targets)
+        self.integration_targets_by_name = dict((t.name, t) for t in self.integration_targets)
+
+        self.posix_integration_by_module = dict((m, t.name) for t in self.integration_targets
+                                                if 'posix/' in t.aliases for m in t.modules)
+        self.windows_integration_by_module = dict((m, t.name) for t in self.integration_targets
+                                                  if 'windows/' in t.aliases for m in t.modules)
+        self.network_integration_by_module = dict((m, t.name) for t in self.integration_targets
+                                                  if 'network/' in t.aliases for m in t.modules)
+
+    def classify(self, path):
+        """
+        :type path: str
+        :rtype: dict[str, str] | None
+        """
+        result = self._classify(path)
+
+        # run all tests when no result given
+        if result is None:
+            return None
+
+        # compile path if eligible
+        if path in self.compile_paths:
+            result['compile'] = path
+
+        # run sanity on path unless result specified otherwise
+        if 'sanity' not in result:
+            result['sanity'] = path
+
+        return result
+
+    def _classify(self, path):
+        """
+        :type path: str
+        :rtype: dict[str, str] | None
+        """
+        filename = os.path.basename(path)
+        name, ext = os.path.splitext(filename)
+
+        minimal = {}
+
+        if path.startswith('.github/'):
+            return minimal
+
+        if path.startswith('bin/'):
+            return minimal
+
+        if path.startswith('contrib/'):
+            return {
+                'units': 'test/units/contrib/'
+            }
+
+        if path.startswith('docs/'):
+            return minimal
+
+        if path.startswith('docs-api/'):
+            return minimal
+
+        if path.startswith('docsite/'):
+            return minimal
+
+        if path.startswith('examples/'):
+            return minimal
+
+        if path.startswith('hacking/'):
+            return minimal
+
+        if path.startswith('lib/ansible/modules/'):
+            module = self.module_names_by_path.get(path)
```

Comment:
Yes, I believe all of the changes for `ansible-test` should work as-is once we drop the `core` and `extras` designations. If you find any place where it looks like that is not the case, let me know.